packages/integrations/node/src/http-server.ts

```diff
@@ -23,12 +25,55 @@ function parsePathname(pathname: string, host: string | undefined, port: number)
 
 export function createServer(
 	{ client, port, host, removeBase }: CreateServerOptions,
-	handler: http.RequestListener
+	handler: http.RequestListener,
+	trailingSlash: AstroUserConfig['trailingSlash']
```

Comment:
I'm not sure that `createServer` is the right place to handle this... Seems like we should be able to read the `trailingSlash` value from the manifest and use that to point requests to the correct route.

Reply:
Yeah sounds right. What I'm unsure about is why this needs to happen in the adapter level and not inside of Astro core itself. It seems like we should be handling trailingSlash ourselves. I can't recall why we do not.

It might be the case that we forgot to initially, and thought it would be backwards incompatible to do so. I do have some recollection of handle trailing slash for things like assets at build time (by making sure it was correct in the manifest).

If that's the case, then I thinking having in the Node adapter is ok, and if we ever fix it in core then it will just already be fixed before it gets there.

In any case, fixing it inside of the middleware seems ideal.